src/core/geometry/qgswkbtypes.h

```diff
@@ -455,6 +455,17 @@ class CORE_EXPORT QgsWKBTypes
       return Unknown;
     }
 
+    /** Returns the modified input geometry type according to hasZ / hasM */
+    static Type zmType( Type type, bool hasZ, bool hasM )
+    {
+      type = singleType( type );
```

Comment:
Can you please add unit tests for this to test_qgsgeometry.py. As the comments in this file say we need 100% test coverage for this class.